test/payments_request_previous_test.go

```diff
@@ -79,214 +93,84 @@ func TestMakeCardVerificationPrevious(t *testing.T) {
-
-	//Customer
-	assert.NotEmpty(t, paymentResponse.Customer)
-	customer := paymentResponse.Customer
-	assert.NotEmpty(t, customer)
-	assert.NotEmpty(t, customer.Id)
-	assert.NotEmpty(t, customer.Name)
-
-	//Processing
-	assert.NotEmpty(t, paymentResponse.Processing)
-	processing := paymentResponse.Processing
-	assert.NotEmpty(t, processing)
-	assert.NotEmpty(t, processing.AcquirerTransactionId)
-	assert.NotEmpty(t, processing.RetrievalReferenceNumber)
-
-	//Risk
-	assert.False(t, paymentResponse.Risk.Flagged)
+	cases := []struct {
+		name           string
+		paymentRequest abc.PaymentRequest
+		checker        func(*abc.PaymentResponse, error)
+	}{
+		{
+			name:           "when get a payment card source request then return a response",
+			paymentRequest: paymentRequest,
+			checker: func(response *abc.PaymentResponse, err error) {
+				assert.Nil(t, err)
+				assert.NotNil(t, response)
+
+				assert.NotEmpty(t, response.Id)
+				assert.NotEmpty(t, response.ProcessedOn)
+				assert.NotEmpty(t, response.Reference)
+				assert.NotEmpty(t, response.ActionId)
+				assert.NotEmpty(t, response.ResponseCode)
+				assert.NotEmpty(t, response.SchemeId)
+				assert.NotEmpty(t, response.ResponseSummary)
+				assert.Equal(t, payments.CardVerified, response.Status)
+				assert.Equal(t, int64(0), response.Amount)
+				assert.True(t, response.Approved)
+				assert.NotEmpty(t, response.AuthCode)
+				assert.NotEmpty(t, response.Currency)
+				assert.Nil(t, response.ThreeDs)
+
+				//Source
+				assert.NotEmpty(t, response.Source)
+				responseCardSource := response.Source.ResponseCardSource
+				assert.NotEmpty(t, payments.CardSource, responseCardSource.Type)
+				assert.NotEmpty(t, responseCardSource.Id)
+				assert.NotEmpty(t, responseCardSource.AvsCheck)
+				assert.NotEmpty(t, responseCardSource.CvvCheck)
+				assert.NotEmpty(t, responseCardSource.Bin)
+				assert.NotEmpty(t, responseCardSource.ExpiryYear)
+				assert.NotEmpty(t, responseCardSource.ExpiryMonth)
+				assert.NotEmpty(t, responseCardSource.Last4)
+				assert.NotEmpty(t, responseCardSource.Name)
+				assert.NotEmpty(t, responseCardSource.FastFunds)
+				assert.NotEmpty(t, responseCardSource.Fingerprint)
+				assert.True(t, responseCardSource.Payouts)
+
+				//Customer
+				assert.NotEmpty(t, response.Customer)
+				customer := response.Customer
+				assert.NotEmpty(t, customer)
+				assert.NotEmpty(t, customer.Id)
+				assert.NotEmpty(t, customer.Name)
+
+				//Processing
+				assert.NotEmpty(t, response.Processing)
+				processing := response.Processing
+				assert.NotEmpty(t, processing)
+				assert.NotEmpty(t, processing.AcquirerTransactionId)
+				assert.NotEmpty(t, processing.RetrievalReferenceNumber)
+
+				//Risk
+				assert.False(t, response.Risk.Flagged)
+
+				//Links
+				assert.NotEmpty(t, response.Links["self"])
+				assert.NotEmpty(t, response.Links["actions"])
+				assert.Empty(t, response.Links["capture"])
+				assert.Empty(t, response.Links["void"])
```

Comment:
I don't know if this ones are necessary, but maybe we could create new functions to do this assertions and make this cases more readable. Like: assertCustomer, assertProccessing, etc